lib/bandit/extractor.ex

```diff
@@ -27,15 +28,18 @@ defmodule Bandit.Extractor do
             required_length: 0,
             mode: :header_parsing,
             max_frame_size: 0,
-            frame_parser: nil
+            frame_parser: nil,
+            primitive_ops_module: nil
 
   @spec new(module(), Keyword.t()) :: t()
   def new(frame_parser, opts) do
     max_frame_size = Keyword.get(opts, :max_frame_size, 0)
+    primitive_ops_module = Keyword.fetch!(opts, :primitive_ops_module)
```

Comment:
Rather than the `put_new` elsewhere, is there a reason *not* to use `Keyword.get` with a default of `Bandit.PrimitiveOps.WebSocket` here?

Reply:
I thought we don't want to couple the Extractor module with websocket specifics (even though it is only used for websockets as of now).

But we can either do a `Keyword.get/3` as you suggested or add an argument to the new/3 so it looks like `def new(frame_parser, primitive_ops_module, opts)`. LMK what you prever.

Reply:
Good point about extractor reuse!

It's a required thing, so I think a separate arg to new/3 makes the most sense

Reply:
Updated. Also do you think I should drop `_module` suffix from `primitive_ops_module`? 🤔